instance/models/utils.py

```diff
@@ -44,3 +55,263 @@ def save(self, *args, **kwargs):
         """Call :meth:`full_clean` before saving."""
         self.full_clean()
         super(ValidateModelMixin, self).save(*args, **kwargs)
+
+
+class ClassProperty(property):
+    """ Same as built-in 'property' global but also works when accessed as a class attribute """
+    def __get__(self, cls, owner):
+        return self.fget.__get__(None, owner)()  # pylint: disable=no-member
+
+
+class ResourceState:
+    """
+    A [finite state machine] state class representing the overall state of a resource,
+    such as a server, a database, a DNS entry, etc.
+    """
+    # state_id: A string uniquely identifying this state
+    state_id = None
+
+    @ClassProperty
```

Comment:
People generally use meta-classes to support properties on classes, but I actually prefer this decorator for this use case.